src/RIEStatefulBase.js

```diff
@@ -13,11 +13,15 @@ export default class RIEStatefulBase extends RIEBase {
 
     finishEditing = () => {
         let newValue = ReactDOM.findDOMNode(this.refs.input).value;
-        this.doValidations(newValue);
-        if(!this.state.invalid && this.props.value !== newValue) {
+        const result = this.doValidations(newValue);
+        if(result && this.props.value !== newValue) {
             this.commit(newValue);
         }
-        this.cancelEditing();
+        if(!result && this.props.handleValidationFail) {
```

Comment:
ideally, `this.doValidation` should return `false` when no error and `error message` when validation fails. But that will change current `validate` interface.

a workaround is, given the `newValue`, re-validate it and get the error message.